source/common/http/http1/codec_impl.cc

```diff
@@ -539,39 +558,54 @@ Http::Status ConnectionImpl::innerDispatch(Buffer::Instance& data) {
   return Http::okStatus();
 }
 
-size_t ConnectionImpl::dispatchSlice(const char* slice, size_t len) {
+Envoy::StatusOr<size_t> ConnectionImpl::dispatchSlice(const char* slice, size_t len) {
+  ASSERT(codec_status_.ok());
   ssize_t rc = http_parser_execute(&parser_, &settings_, slice, len);
+  if (!codec_status_.ok()) {
```

Comment:
Should I wrap this if in to the if statement below? I wanted to *exactly* preserve execution, but this could easily be handled in the statement below.

Reply:
I think it's probably fine to fold into below but up to you.

Reply:
Earlier we didnt `sendProtocolError` on a thrown exception. I assume we might want to?